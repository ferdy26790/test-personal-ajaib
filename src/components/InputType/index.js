import * as React from 'react';
import TextField from '@mui/material/TextField';
import { MenuItem } from '@mui/material';
import { debounce } from 'lodash';

export const FilterInput = ({label, handleChange, value, options}) => {
    return <TextField
            select
            id="outlined-search"
            label={label}
            style={{
            width:'180px'
            }}
            value={value}
            onChange={(e) => {
                handleChange(e.target.value)
            }}
            >
                {options.map((o) => <MenuItem key={o.value} value={o.value}>{o.label}</MenuItem>)}
            </TextField>
}

export const InputText = ({label, handleChange}) => {
    const debounceHandleChange = React.useCallback(
        debounce(handleChange, 300)
    ,[])
    return <TextField
    id="outlined-search"
    className="test"
    label={label}
    onChange={debounceHandleChange}
  />
}

export default function InputType({type, label, handleChange, value, options}) {
  const INPUT_TYPE_COMPONENTS = {
    select: <FilterInput label={label} handleChange={handleChange} options={options} value={value}/>,
    text: <InputText label={label} handleChange={handleChange} value={value}/>
  }
  return INPUT_TYPE_COMPONENTS[type] || null
}
