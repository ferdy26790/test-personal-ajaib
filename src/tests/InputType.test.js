
import InputType, { FilterInput, InputText } from '../components/InputType';
import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import '../setupTests';
import { TextField } from '@mui/material';
import debounce from 'lodash/debounce';

// Tell Jest to mock this import
jest.mock('lodash/debounce');

const colorOptions = [
    {
        label: "Red",
        value: "red"
    },
    {
        label: "Green",
        value: "green"
    },
    {
        label: "Blue",
        value: "blue"
    }
]
jest.useFakeTimers();
describe('<InputType />', () => {
    it("Should render input text by given type correctly", () => {
        const spy = jest.fn();
        const wrapper = shallow(<InputType type="text" label="search" value="test" handleChange={() => {spy}} />);
        expect(wrapper.prop('label')).toEqual("search");
        expect(wrapper.containsAnyMatchingElements(<InputText />))

    });

    it("Should render input text by given type correctly", () => {
        const spy = jest.fn();
        const wrapper = shallow(
            <InputType type="select"
                       label="color"
                       options={colorOptions}
                       value="green" 
                       handleChange={() => {jest.fn()}} />
        );
        expect(wrapper.prop('label')).toEqual("color");
        expect(wrapper.containsAnyMatchingElements(<FilterInput />))

    });

    it("Should trigger handleChange method", () => {
        const spy = jest.fn();
        // debounce.mockImplementation(spy)
        const wrapper = mount(<InputType type="text" label="search" value="test" handleChange={() => {spy}} />);
        // <InputText label="search" value="test" handleChange={() => {spy}} />);
        expect(wrapper.find(InputText).find(TextField).render().find('input')).toHaveLength(1);
        console.log("-------- inputtt", wrapper.find(InputText).find(TextField).find('input').target);
        
        wrapper.find(InputText).find(TextField).find('input').simulate('change', { target: { value: "Claire" } });
        // jest.runAllTimers(300);
        
        // expect(spy).toHaveBeenCalled();

    });


});

