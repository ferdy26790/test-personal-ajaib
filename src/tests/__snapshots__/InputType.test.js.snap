// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputType /> Should render filter input by given type correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  style={
    Object {
      "width": "180px",
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    htmlFor="outlined-search"
    id="outlined-search-label"
  >
    color
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
    variant="outlined"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="outlined-search-label outlined-search"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="outlined-search"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Green
    </div>
    <input
      aria-hidden={true}
      autoFocus={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="green"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1in441m"
      >
        <span>
          color
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`<InputType /> Should render input text by given type correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="outlined-search"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
